core/domain/exp_jobs_one_off_test.py

```diff
@@ -290,6 +292,132 @@ def test_first_published_time_of_exploration_that_is_unpublished(self):
+        """Test that if the same user makes more than one commit that changes
+        the content of an exploration, the user is only represented once in the
+        list of contributors for that exploration.
+        """
+        self.signup(self.EMAIL_A, self.USERNAME_A)
+        self.user_a_id = self.get_user_id_from_email(self.EMAIL_A)
+
+        # have one user make two commits
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.user_a_id, title='Original Title')
+        exploration_model = exp_models.ExplorationModel.get(
+            self.EXP_ID, strict=True, version=None)
+        exploration_model.title = 'New title'
+        exploration_model.commit(
+            self.user_a_id, 'Changed title.', [])
+
+        # Run the job to compute the contributor id's.
+        job_id = exp_jobs_one_off.ExpSummariesContributorsOneOffJob.create_new()
+        exp_jobs_one_off.ExpSummariesContributorsOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        # Verify that the length of the contributor list is one, and that
+        # the list contains the user who made these commits.
+        exploration_summary = exp_services.compute_summary_of_exploration(exploration)
+        self.assertEqual(len(exploration_summary.contributor_ids), 1)
+        self.assertEqual(
+            [self.user_a_id], exploration_summary.contributor_ids)
+
+    def test_contributors_with_only_reverts_not_counted(self):
+        """Test that contributors who have only done reverts do not
+        have their user id appear in the contributor list.
+        """
+        # sign up two users
+        self.signup(self.EMAIL_A, self.USERNAME_A)
+        self.user_a_id = self.get_user_id_from_email(self.EMAIL_A)
+        self.signup(self.EMAIL_B, self.USERNAME_B)
+        self.user_b_id = self.get_user_id_from_email(self.EMAIL_B)
+        # have one user make two commits
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.user_a_id, title='Original Title')
+        exploration_model = exp_models.ExplorationModel.get(
+            self.EXP_ID, strict=True, version=None)
+        exploration_model.title = 'New title'
+        exploration_model.commit(
+            self.user_a_id, 'Changed title.', [])
+
+        # Have the second user revert version 2 to version 1
+        exp_services.revert_exploration(self.user_b_id, self.EXP_ID, 2, 1)
+
+        # Run the job to compute the contributor id's.
+        job_id = exp_jobs_one_off.ExpSummariesContributorsOneOffJob.create_new()
+        exp_jobs_one_off.ExpSummariesContributorsOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        # Verify that the committer list does not contain the user
+        # who only reverted.
+        exploration_summary = exp_services.compute_summary_of_exploration(exploration)
+        self.assertNotIn(self.user_b_id, exploration_summary.contributor_ids)
```

Comment:
replaced with just equality-- seemed like similar situation to above, where equality entails that user_b_id is not in the contributors list.